wsi_superpixel_guided_labeling/web_client/views/vue/components/ActiveLearningSetup/ActiveLearningInitialLabels.vue

```diff
@@ -229,13 +233,14 @@ export default Vue.extend({
             const index = boundaries ? (event.index - event.index % 2) : event.index;
             const offset = boundaries ? 1 : 0;
             const data = this.overlayLayer.data();
-            // the +1 accounts for the default, reset to default if already labeled
-            const categoryIndex = data[index] === 0 ? this.categoryIndex + 1 : 0;
-            data[index] = data[index + offset] = categoryIndex;
+            const previousLabel = data[index];
+            // the +1 accounts for the default, reset to default if already labeled with the selected category
+            const newLabel = previousLabel !== this.categoryIndex + 1 ? this.categoryIndex + 1 : 0;
```

Comment:
Previously if this superpixel was labeled and we clicked on it, we set it to default, regardless of what category is currently active for labeling. This was unintuitive to use, since you'd need 2 clicks to change an existing label.

Reply:
Good.